### ISSUE TITLE
Contact Form Block: Add `Contact Form` Keyword

### ISSUE THREAD

**Issue by JoshuaGoode** [Automattic/jetpack#14000]
#### Is your feature request related to a problem? Please describe.
For finding blocks, I personally use the search tool and guide others through doing so as well. With an emphasis on search, ensuring blocks have the proper keywords tag is important.

if a user searches “Contact Form” — no results will appear on a vanilla WPCOM site. To find the form, they have to limit their search to `Contact` or `Form`.

![image](https://user-images.githubusercontent.com/4887830/68493866-4b5d4e00-021b-11ea-9349-6a68ee2fa4cc.png)

This is because the Contact Form block is titled `Form` and is limited to the following keywords:
`email`
`feedback`
`contact`

See: https://github.com/Automattic/jetpack/blob/master/extensions/blocks/contact-form/index.js#L29

#### Describe the solution you'd like

Add `contact form` as a keyword to the Contact Form block.

**Comment by jeherve** [comment]
Unfortunately we can't use more than 3 keywords right now, so we'd have to remove an existing keyword to add that one. 

Do we really need that new keyword though? Wouldn't one see the Form block as they're starting to type "contact":

![Screen Recording 2019-11-08 at 05 41 PM](https://user-images.githubusercontent.com/426388/68494727-34385380-024f-11ea-8044-35c2115dde07.gif)

cc'ing @Automattic/jetpack-design to get their opinion on this.

**Comment by keoshi** [comment]
I agree with @jeherve — while typing `contact` you're most likely to see the blocks being filtered with every character added, so I'd argue adding `contact form` wouldn't make much of a difference.

One thing we can point out to the Gutenberg developers is that search should possibly also take into consideration the block's name.

**Comment by jeherve** [comment]
> One thing we can point out to the Gutenberg developers is that search should possibly also take into consideration the block's name.

That sounds like a good idea. @JoshuaGoode Do you think you could open an issue in the Gutenberg repo about this?

Thanks!

**Comment by keoshi** [comment]
I asked the Gutenberg team and @mtias highlighted an important point: Gutenberg does indeed use the block's name for search purposes. The issue here is that the block is called `Form`, and looking up `Contact Form` is the same as `Contact (AND) Form` so nothing will be returned.

We can fix this on Jetpack by renaming the block `Contact Form`. That would satisfy this issue and give the block a broader range of search possibilities.

**Comment by jeherve** [comment]
> We can fix this on Jetpack by renaming the block Contact Form.

That's how the form was originally named, but we changed the name to just form in https://github.com/Automattic/wp-calypso/pull/28734

This was done as the form block had the opportunity to be more than just a contact form block.

cc'ing @MichaelArestad who worked on that rename back then.

**Comment by keoshi** [comment]
If that still stands, then @JoshuaGoode's suggestion sounds good to me. At the very least it will help the block get visibility that it possibly lost since that change.

I'd still argue that most people would tend to better grasp `Contact Form` as the name of the block, even if its use cases are broader than that.

**Comment by jeherve** [comment]
> then @JoshuaGoode's suggestion sounds good to me.

Which keyword would you remove and replace with `contact form`?

**Comment by mtias** [comment]
"contact" => "contact form"?

**Comment by keoshi** [comment]
I'd replace `contact` for `contact form`, which is in line with the explanations above.

**Comment by jeherve** [comment]
👍. Happy to quickly throw together a PR if everyone is happy with that.

**Comment by keoshi** [comment]
I'd still like to hear @MichaelArestad's thoughts on the block's naming and whether we should update the keyword or the name?

**Comment by MichaelArestad** [comment]
We changed it for a couple reasons. The first is that "Contact form" has specific context where this form can be used for quite a range of things. The second is brevity. It's short and looks good in the editor and inserter.

I was also pretty sure we had added this as a keyword, but I guess not. Let's do it.

**Comment by keoshi** [comment]
Thanks for the context, @MichaelArestad !

@jeherve Do you want to take this one or should I?

Finally, thanks again for bringing this up @JoshuaGoode !

**Comment by JoshuaGoode** [comment]
Whoa, I stepped away for a bit and didn't expect all this movement, haha. 

Thanks for reviewing this, all! @MichaelArestad @keoshi @jeherve 

`contact form` instead of just `contact` sounds great!  I've encountered multiple situations where users have typed out `Contact Form` and ignored the Form block that appeared as they were typing things out.

I've discussed this in the past but never took the time to open an issue or submit a PR. Also sparked by recent internal discussion.

Ref: p9Jlb4-13I-p2
